conda/cli/main_search.py

```diff
@@ -78,26 +80,22 @@ def configure_parser(sub_parsers):
         default=None,
     )
     p.add_argument(
-        "--spec",
-        action="store_true",
-        help="""Treat the regex argument as a package specification instead
-        (package_name[=version[=build]]).""",
+        'spec',
```

Comment:
Let's call this `match_spec` instead of `spec` so we can keep the `--spec` flag as a no-op.